src/main/kotlin/com/fasterxml/jackson/module/kotlin/ReflectionCacheNew.kt

```diff
@@ -0,0 +1,71 @@
+package com.fasterxml.jackson.module.kotlin
+
+import com.fasterxml.jackson.databind.introspect.AnnotatedConstructor
+import com.fasterxml.jackson.databind.introspect.AnnotatedMethod
+import com.fasterxml.jackson.databind.introspect.AnnotatedWithParams
+import com.fasterxml.jackson.databind.util.LRUMap
+import java.lang.reflect.Constructor
+import java.lang.reflect.Method
+import kotlin.reflect.full.extensionReceiverParameter
+import kotlin.reflect.full.instanceParameter
+import kotlin.reflect.jvm.kotlinFunction
+
+// TODO: Consider whether it is necessary to include it in the ReflectionCache class.
```

Comment:
As for the caching and fetching, should it be integrated into `ReflectionCache`?
Also, if I do, should I cache both the `KFunction` cache and the `Instantiator` cache?

Reply:
Merging caches may make sense, one less thing to carry around. Not sure about latter, not familiar enough with codebase to have opinion.

Reply:
After re-reading the code again, I came to the conclusion that it should be integrated into `ReflectionCache`.

The conversion from `Java` `Constructor`/Method` to `KFunction` is cached, so it would be better to use the cache when generating the `Instantiator`.
The fix has already been pushed.
c320833

This comment also resolves.